src/uct/ib/mlx5/ib_mlx5_dv.c

```diff
@@ -33,6 +33,10 @@ static ucs_status_t uct_ib_mlx5_device_init(uct_ib_device_t *dev)
     struct ibv_cq *cq;
     struct ibv_qp *qp;
 
+    if (!(uct_ib_device_spec(dev)->flags & UCT_IB_DEVICE_FLAG_MLX5_PRM)) {
```

Comment:
@Artemy-Mellanox this means we would not be able to remove the device table in upstream driver, right? Is there something that need to fix in rdma-core/MLNX_OFED? If yes pls open internal RM issue for Majd

Reply:
AFAIU we should call mlx5dv* functions only if we know there is mlx5 provider behind. I still don't know what is the correct way to check this, so using what we have currently.

Reply:
i think that as general API, mlx5dv should be provide a query-if-supported functionality.
@jgunthorpe what do you think?

Reply:
There is a gap in the DV definition, there is no way for the user to know if the DV works with the ibv_context, and calling DV functions with a mismatched context will crash. This is something we really should address..